src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/r/SignatureToolTipManager.java

```diff
@@ -81,9 +145,45 @@ public void initialize(UIPrefs uiPrefs,
       server_ = server;
    }
    
+   private void attachPreviewHandler()
+   {
+      detachPreviewHandler();
+      preview_ = Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent preview)
+         {
+            if (preview.getTypeInt() == Event.ONMOUSEMOVE)
+            {
+               NativeEvent event = preview.getNativeEvent();
+               coordinates_ = new MouseCoordinates(
+                     event.getClientX(),
+                     event.getClientY());
+            }
+            else if (preview.getTypeInt() == Event.ONKEYDOWN)
+            {
+               coordinates_ = null;
+            }
+            
+            timer_.schedule(DELAY_MS);
```

Comment:
I think that could get very expensive as there are potentially thousands of
them in short periods of time.

On Tue, Mar 1, 2016 at 5:27 PM, Jonathan <notifications@github.com> wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/r/SignatureToolTipManager.java
> <https://github.com/rstudio/rstudio/pull/614#discussion_r54647101>:
>
> > +         @Override
> > +         public void onPreviewNativeEvent(NativePreviewEvent preview)
> > +         {
> > +            if (preview.getTypeInt() == Event.ONMOUSEMOVE)
> > +            {
> > +               NativeEvent event = preview.getNativeEvent();
> > +               coordinates_ = new MouseCoordinates(
> > +                     event.getClientX(),
> > +                     event.getClientY());
> > +            }
> > +            else if (preview.getTypeInt() == Event.ONKEYDOWN)
> > +            {
> > +               coordinates_ = null;
> > +            }
> > +
> > +            timer_.schedule(DELAY_MS);
>
> do we want to schedule the timer for *all* types of native events?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/614/files#r54647101>.
>